setup.py

```diff
@@ -22,7 +22,7 @@
     # Versions should comply with PEP440.  For a discussion on single-sourcing
     # the version across setup.py and the project code, see
     # https://packaging.python.org/en/latest/single_source_version.html
-    version='0.4.0',
+    version='0.5.0rc2',
```

Comment:
We need to bump this whenever we want to tag a new release.

Note that there's [other ways to manage this](https://packaging.python.org/en/latest/guides/single-sourcing-package-version/) but this seemed fine for now.

Reply:
The issue with this is that we will then need to ensure that we update the version with every PR otherwise we will get an error since the build with try to publish it on merge from a PR. It would be nice if we could get the github release version and then use that instead though I guess its fine if we dont for now.

Reply:
Yeah I was considering using the https://pypi.org/project/setuptools-scm/ package that's linked there but I didn't wanna mess with it for now. I actually set it up so that it only tries to publish the package if a tag is pushed and not just any merges, so we should be fine for now. I ran into the issue with TestPyPI because even with TestPyPI it prevents you from uploading the same package with the same version.